fuse/fuse_test.go

```diff
@@ -17,104 +19,290 @@ import (
+	defer os.RemoveAll(bfInfo.dbPath)
+	// first unmount fuse directory
+	if err := lazyUnmount(bfInfo.fuseMount.Dir); err != nil {
+		skipableErr := "exit status 1: fusermount: entry for " + bfInfo.fuseMount.Dir + " not found in /etc/mtab"
+		log.Debug(skipableErr)
+		if err.Error() != skipableErr {
+			return &nothing{}, err
+		}
+	}
 
+	// now close brig FS
+	err := bfInfo.bfs.Close()
+	if err != nil {
+		log.Fatalf("Could not close brig filesystem: %v", err)
+	}
+	return &nothing{}, err
+}
+
+func makeFuseMount(bfs *catfs.FS, mntPath string, opts MountOptions) (*Mount, error) {
 	// Make sure to unmount any mounts that are there.
 	// Possibly there are some leftovers from previous failed runs.
-	lazyUnmount(mntPath)
+	if err := lazyUnmount(mntPath); err != nil {
+		skipableErr := "exit status 1: fusermount: entry for " + mntPath + " not found in /etc/mtab"
+		log.Debug(skipableErr)
+		if err.Error() != skipableErr {
+			return nil, err
+		}
+	}
 
 	if err := os.MkdirAll(mntPath, 0777); err != nil {
-		t.Fatalf("Unable to create empty mount dir: %v", err)
+		log.Fatalf("Unable to create empty mount dir: %v", err)
+		return nil, err
 	}
 
-	defer testutil.Remover(t, mntPath)
-
-	mount, err := NewMount(fs, mntPath, nil, opts)
+	mount, err := NewMount(bfs, mntPath, nil, opts)
 	if err != nil {
-		t.Fatalf("Cannot create mount: %v", err)
+		log.Fatalf("Cannot create mount: %v", err)
+		return nil, err
 	}
+	return mount, err
+}
+
+type brigPayload struct {
+	Path string
+	Data []byte
+}
+
+func (bmh *brigHelp) brigStage(ctx context.Context, req brigPayload) (*nothing, error) {
+	err := bfInfo.bfs.Stage(req.Path, bytes.NewReader(req.Data))
+	return &nothing{}, err
+}
 
-	f(mount)
+// Get data from a file stored by brig fs
+func (bmh *brigHelp) brigGetData(ctx context.Context, req brigPayload) (*brigPayload, error) {
+	out := brigPayload{}
+	out.Path = req.Path
 
-	if err := mount.Close(); err != nil {
-		t.Fatalf("Closing mount failed: %v", err)
+	stream, err := bfInfo.bfs.Cat(req.Path)
+	if err != nil {
+		log.Fatalf("Could not get stream for a brig file: %v", err)
+		return &out, err
 	}
+	result := bytes.NewBuffer(nil)
+	_, err = stream.WriteTo(result)
+	if err != nil {
+		log.Fatalf("Streaming to a buffer failed: %v", err)
+		return &out, err
+	}
+	out.Data = result.Bytes()
+
+	return &out, err
 }
 
-func withMount(t *testing.T, opts MountOptions, f func(mount *Mount)) {
-	withDummyFS(t, func(fs *catfs.FS) {
-		withMountFromFs(t, opts, fs, f)
-	})
+var helpers spawntest.Registry
+var brigHelper = helpers.Register("brigHelp", &brigHelp{})
 
+type mountInfo struct { // fuse related info available to OS layer
+	Dir  string
+	Opts MountOptions
 }
 
-func checkForCorrectFile(t *testing.T, path string, data []byte) {
-	// Try to read it over fuse:
-	helloBuffer := &bytes.Buffer{}
-	fd, err := os.Open(path)
+func withMount(t *testing.T, opts MountOptions, f func(ctx context.Context, control *spawntest.Control, mount *mountInfo)) {
```

Comment:
I saw that you often pass the zero value of MountOptions to ``withMount()``. It's common practice to make ``opts`` a pointer and fill the defaults in ``withMounts`` if it's nil. Just a small nitpick, but makes code a bit tidier I think.

Reply:
I would like to keep it this way. Otherwise I had to constantly check the definition of the function to see what place the nil is taking.

My way is good for copy-pasting a test header, I see exactly what is sent to withMount. It is more typing if you do it from scratch, but who does it?

Am I missing something?

Reply:
I'm not opposed to leaving it as it is. I just took it as an example for a typical Go-technique. So I will approve both ways - it's more of a philosophical question after all.

> Otherwise I had to constantly check the definition of the function to see what place the nil is taking.

Fair point, but usually good text editors IDEs help with that. If I don't remember what e.g. the third parameter I just jump to the function definition. If I always have to do it, it's usually a sign for me to simplify the function because it might have too many parameters. Arguably languages like python make this easier by the optional use of keyword arguments like ``f(buffer_size=4)``. 

> ...but who does it?

Me often, when I'm unfamiliar with some code, I try to avoid copy&pasting. Typing function calls makes you think. Later on I do plenty of copy&paste too, though. Anyway, APIs should be designed to be easy to be used and understood, not to be easy to copy & paste.

Reply:
I left it as it were. But it is time for me to tune up my Vim